### PR TITLE
Start of BSV support

### PR DESCRIPTION
**Pull request by dfoderick**
Not ready to merge but needs discussion on following points
(1) BSV does not have any site that supports `/rawblock` yet. And given that some blocks will be huge (I think largest one is 100K?) is this method necessary or is it a nice-to-have feature?
(2) Testnet on bsv is very limited. I don't think any bsv testnet insight api will be supported in the near future. We should inquire about testing on the Scaling Test Network (STN)

Remaining work  
* move test data to a separate file
* enable sendTransaction test
* Verify that utxo test is correct

### CODE REVIEW

**Comment by ClemensLey** [comment]
Sounds good. We are not using rawBlock in the BitcoinToken project right now, so to me it is a nice-to-have but not necessary.

If testnet on BCV is too limited then I'd recommend we just use the testnet for BCH right now.

The code looks great so far. Will leave this open for now until it's ready to be merged

**Comment by brentongunning** [comment]
Hey Dave, awesome progress. I see you added checks on the responses, and they look great. Got a few Qs about postman and the tests. Would it be easiest to review this together over a call this week?

**Comment by dfoderick** [comment]
> Hey Dave, awesome progress. I see you added checks on the responses, and they look great. Got a few Qs about postman and the tests. Would it be easiest to review this together over a call this week?

Yes. I will contact you. I've taken the task about as far as I can go right now. Let's do a code review and finish off the task.

File: src/apibase.js [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import { IApi, IApiMinimal } from './api'
+import ApiError from './error'
+import { Address, Transaction } from 'bitcoinsource'
+import type { OutputId, TransactionId, Txo } from './types'
+import { removeDuplicates, renameProperty, unwrapAxiosResponse } from './util'
+
+/**
+ * Base class for implementing Api
+ */
+export class ApiBaseMinimal implements IApiMinimal {
```
Comment (brentongunning) [inline src/apibase.js:16]: Maybe call this InsightApiBase/Minimal? We might have other APIs someday that fit into the general API interface.

**Pull request merged**